contracts/ReaperAutoCompound-Liquid-v2-Beethoven.sol

```diff
@@ -205,22 +207,40 @@ contract ReaperAutoCompound_LiquidV2_Beethoven is ReaperBaseStrategy {
      * @dev Request {bpToken} to the {BEET_VAULT} based on underlying tokens balances
      */
     function _addLiquidity() internal {
-        //todo update to fit strategy
-        //      swap correct ratios of wftm to underlying tokens
-        //      allow the vault to transfer underlying tokens
-        //      join pool with tokens and get bpToken
+        uint256 wftmBal = IERC20(WFTM).balanceOf(address(this));
+
+        for (uint256 i; i < totalUnderlyingTokens; i++) {
+            address token = bptUnderlyingTokens[i];
+            uint256 wftmToSwap = (wftmBal * underlyingToWeight[token]) / PERCENT_DIVISOR;
+            wftmBal -= wftmToSwap;
```

Comment:
All calc should be based on the same number, and wftmbal should be updated once all tokens have been swapped